lib/factory_bot/factory.rb

```diff
@@ -84,6 +84,7 @@ def compile
       unless @compiled
         parent.compile
         parent.defined_traits.each { |trait| define_trait(trait) }
+        @definition.expand_enum_traits(build_class)
         @definition.compile
```

Comment:
In an ideal world we could pass the `build_class` to `@definition.compile` and let the definition itself handle expanding the traits, but we call `@definition.compile` in other places where a `build_class` is not available.

Reply:
i agree with placing the triggering logic in `@definition.compile` would be better, but i also agree that it may not worth it to refactor other parts largely only for this.

Reply:
I pushed up 9c4d7c5, which solves this but adds an extra conditional to `Definition#compile`. Do you think that is worth it?